app/src/main/java/com/example/mac/chartr/activities/PostTripActivity.java

```diff
@@ -37,10 +44,127 @@
+    private void initPickers() {
+    /*
+    SETUP DEPARTURE DATE PICKER
+     */
+        departureEditText = (EditText) findViewById(R.id.editTextDepartureDate);
+        departureDate  = (view, year, monthOfYear, dayOfMonth) -> {
+            departureCalendar.set(Calendar.YEAR, year);
+            departureCalendar.set(Calendar.MONTH, monthOfYear);
+            departureCalendar.set(Calendar.DAY_OF_MONTH, dayOfMonth);
+            updateDepartureDateLabel();
+        };
+        departureEditText.setOnClickListener(v -> {
+            new DatePickerDialog(PostTripActivity.this, departureDate, departureCalendar
+                    .get(Calendar.YEAR), departureCalendar.get(Calendar.MONTH),
+                    departureCalendar.get(Calendar.DAY_OF_MONTH)).show();
+        });
+
+        /*
+        SETUP RETURN DATE PICKER
+         */
+        returnEditText = (EditText) findViewById(R.id.editTextReturnDate);
+        returnDate  = (view, year, monthOfYear, dayOfMonth) -> {
+            returnCalendar.set(Calendar.YEAR, year);
+            returnCalendar.set(Calendar.MONTH, monthOfYear);
+            returnCalendar.set(Calendar.DAY_OF_MONTH, dayOfMonth);
+            updateReturnDateLabel();
+        };
+        returnEditText.setOnClickListener(v -> {
+            new DatePickerDialog(PostTripActivity.this, returnDate, returnCalendar
+                    .get(Calendar.YEAR), returnCalendar.get(Calendar.MONTH),
+                    returnCalendar.get(Calendar.DAY_OF_MONTH)).show();
+        });
+
+        /*
+        SETUP DEPARTURE TIME PICKER
+         */
+        departureTimeText = (EditText) findViewById(R.id.editTextDepartureTime);
+        departTime  = (TimePicker timePicker, int selectedHour, int selectedMinute) -> {
+            String modifier = "am";
+            String minutes, hours;
+            if (selectedHour > 12) {
+                selectedHour -= 12;
+                modifier = "pm";
+            }
+            minutes = extractTimeString(selectedMinute);
+            hours = extractTimeString(selectedHour);
+
+            departureTimeText.setText("" + hours + ":" + minutes + " " + modifier);
+        };
+        departureTimeText.setOnClickListener(v -> {
+            new TimePickerDialog(PostTripActivity.this, departTime, 0, 0,
+                    false).show();
+        });
+
+        /*
+        SETUP RETURN TIME PICKER
+         */
+        returnTimeText = (EditText) findViewById(R.id.editTextReturnTime);
+        returnTime  = (TimePicker timePicker, int selectedHour, int selectedMinute) -> {
+            String modifier = "am";
+            String minutes, hours;
+            if (selectedHour > 12) {
+                selectedHour -= 12;
+                modifier = "pm";
+            }
+            minutes = extractTimeString(selectedMinute);
+            hours = extractTimeString(selectedHour);
+
+            returnTimeText.setText("" + hours + ":" + minutes + " " + modifier);
```

Comment:
Right now, single digit hours have a 0 in front of them.  Do we want to keep it that way?

Reply:
While technically we could display something different and parse it later, it's easiest to keep the leading 0, since the format is later checked against "MM/dd/yyhh:mm a" by combining the date and time fields.

Reply:
Alright.